src/core/positioning.cpp

```diff
@@ -0,0 +1,203 @@
+/***************************************************************************
+  positioning.cpp - Positioning
+
+ ---------------------
+ begin                : 22.05.2022
+ copyright            : (C) 2022 by Mathieu Pellerin
+ email                : mathieu at opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "bluetoothreceiver.h"
+#include "internalreceiver.h"
+#include "positioning.h"
+#include "positioningutils.h"
+
+Positioning::Positioning( QObject *parent )
+  : QObject( parent )
+{
+}
+
+void Positioning::setActive( bool active )
+{
+  if ( mActive == active )
+    return;
+
+  mActive = active;
+
+  if ( mActive )
+  {
+    if ( !mReceiver )
+    {
+      setupDevice();
+    }
+    mReceiver->connectDevice();
+  }
+  else
+  {
+    if ( mReceiver )
+    {
+      mReceiver->disconnectDevice();
```

Comment:
Do we need to set mreceiver to nullptr here to be sure setupdevice is called again (line 38)?

Reply:
@m-kuhn , no we don't, we don't want to re-create the receiver every time someone toggles the positioning on/off. We take care of changing the setting up the device whenever the device name changes, that's the only time we'd need to delete the receiver.